conjure-macros/src/lib.rs

```diff
@@ -160,8 +162,239 @@ pub fn conjure_client(attr: TokenStream, item: TokenStream) -> TokenStream {
     client::generate(attr, item)
 }
 
+/// Creates a Conjure service type wrapping types implementing the annotated trait.
+///
+/// For a trait named `MyService`, the macro will create a type named `MyServiceEndpoints` which
+/// implements the conjure `Service` trait.
+///
+/// # Parameters
+///
+/// The trait can optionally be declared generic over the request body and response writer types by
+/// using the `#[request_body]` and `#[response_writer]` annotations on the type parameters.
+///
+/// # Endpoints
+///
+/// Each method corresponds to a separate HTTP endpoint, and is expected to take `&self` and return
+/// `Result<T, Error>`. Each must be annotated with `#[endpoint]`, which has several parameters:
+///
+/// * `method` - The HTTP method (e.g. `GET`). Required.
+/// * `path` - The HTTP path template. Path parameters should be identified by `{name}` and must
+///     make up an entire path component. Required.
+/// * `produces` - A type implementing `SerializeResponse` which will be used to convert the value
+///     returned by the method into a response. Defaults to `EmptyResponseSerializer`.
+///
+/// Each method argument must have an annotation describing the type of parameter. One of:
+///
+/// * `#[path]` - A path parameter. The path template mut contain a parameter component matching
+///     the argument name.
+///
+///     Parameters:
+///     * `decoder` - A type implementing `DecodeParam` which will be used to decode the value.
+///         Defaults to `FromStrDecoder`.
+///     * `safe` - If set, the parameter will be added to the `SafeParams` response extension.
+/// * `#[query]` - A query parameter.
+///
+///     Parameters:
+///     * `name` - The string used as the key in the encoded URI. Required.
+///     * `decoder` - A type implementing `DecodeParam` which will be used to decode the value.
+///         Defaults to `FromStrDecoder`.
+///     * `safe` - If set, the parameter will be added to the `SafeParams` response extension.
+/// * `#[auth]` - A `BearerToken` used to authenticate the request.
```

Comment:
Maybe worth mentioning that a client must send this as an `Authorization` header with value `Bearer <bearer-token>`

Reply:
It's mentioned down below in the parameter docs

Reply:
Ah, missed that that was for the same field. Makes sense